api/app/Controllers/Http/ChunkedUploadController.js

```diff
@@ -0,0 +1,99 @@
+"use strict";
+const Env = use("Env");
+const path = require("path");
+const fs = require("fs");
+const os = require("os");
+
+const TIMEOUT = 10*60000; // Timeout in milliseconds (30 seconds)
+
+class ChunkedUploadController {
+  deletePartialStoredChunks(totalChunks) {
+    const uploadPath = os.tmpdir();
+    for (let i = 0; i < totalChunks; i++) {
+      const tmpFilePath = path.join(uploadPath, `tempfile${index}.bin`);
+      fs.unlinkSync(tmpFilePath); // Remove individual chunk files
+    }
+  }
+
+  async upload({ request, response, session }) {
+    const { chunk } = request.post();
+    const { chunkNumber, totalChunks, filename } = request.only([
+      "chunkNumber",
+      "totalChunks",
+      "filename",
```

Comment:
The filename we receive from the client can be suffixed to the end of a filename derived from the session. See above.